packages/blade/src/components/Box/BaseBox/getBaseBoxStyles.ts

```diff
@@ -1,102 +1,239 @@
 
-// allows 'auto' as spacing value
-const getSpacingValue = <SpacingValue extends string | number | undefined>({
-  value,
-  theme,
-}: {
-  value: SpacingValue;
-  theme: Theme;
-}): string | undefined => {
-  if (value === 'auto') return 'auto';
-  /**
-   * NOTE: don't allow numbers once we have the scale for sizing
-   * DO NOT PUBLISH THIS COMPONENT PUBLICLY WITH THIS PIECE OF CODE
-   */
-  if (typeof value === 'number') {
-    return makeSpace(value);
+const getResponsiveValue = <T extends string | number | string[]>(
+  value: MakeValueResponsive<T> | undefined,
+  size: keyof Breakpoints = 'base',
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): any | undefined => {
+  if (value === undefined || value === null) {
+    return undefined;
   }
-  return value ? makeSpace(getIn(theme, value)) : undefined;
+
+  if (typeof value === 'string' || typeof value === 'number' || Array.isArray(value)) {
+    if (size === 'base') {
+      return value;
+    }
+
+    // Plain values will already get added in base styles so we don't have to repeat them in media queries
+    return undefined;
+  }
+
+  if (isEmpty(value)) {
+    return undefined;
+  }
+
+  if (isReactNative()) {
+    // In React Native, we map the value `s` token on priority (since the breakpoint maps to mobiles in useBreakpoint hook).
+    // We further look into smaller sizes, then we check base size.
+    // Then we return the first non-undefined value in this priority
+    const priorityArray = [value.s, value.xs, value.base];
+    return priorityArray.find((val) => val !== undefined);
+  }
+
+  return value[size];
+};
+
+const getSpacingValue = (
+  spacingValue:
+    | MakeValueResponsive<SpacingValueType | ArrayOfMaxLength4<SpacingValueType>>
+    | undefined,
+  theme: Theme,
+  size?: keyof Breakpoints,
+): string | undefined => {
+  if (isEmpty(spacingValue)) {
+    return undefined;
+  }
+
+  const responsiveSpacingValue: SpacingValueType | SpacingValueType[] = getResponsiveValue(
+    spacingValue as MakeValueResponsive<string | string[]>,
+    size,
+  );
+
+  if (isEmpty(responsiveSpacingValue)) {
+    return undefined;
+  }
+
+  if (responsiveSpacingValue === 'auto') {
+    return responsiveSpacingValue;
+  }
+
+  if (Array.isArray(responsiveSpacingValue)) {
+    return responsiveSpacingValue.map((value) => getSpacingValue(value, theme)).join(' ');
+  }
+
+  if (typeof responsiveSpacingValue === 'string' && responsiveSpacingValue.startsWith('spacing.')) {
+    const spacingReturnValue = getIn(theme, responsiveSpacingValue);
+    return isEmpty(spacingReturnValue) ? makeSpace(spacingReturnValue) : undefined;
+  }
+
+  // pixel or with unit values
+  return responsiveSpacingValue;
+};
+
+const getBackgroundValue = (
+  backgroundColor: BaseBoxProps['backgroundColor'],
+  theme: Theme,
+  size?: keyof Breakpoints,
+): string => {
+  const responsiveBackgroundValue = getResponsiveValue(backgroundColor, size);
```

Comment:
Yup it will return red.  Calling it here to handle cases like `backgroundColor={{ base: 'red', m: 'yellow' }}`. This function will decide which value to return from here.